README.md

```diff
@@ -104,42 +104,30 @@ class CustomVector {
 
 # Caveats
 
-- If the modules using this library are different, then runtime type information might not match. This will result in runtime errors, `instanceof` checks failing, and undefined behavior. ASON also performs type information validation for objects at the top level, so providing the wrong reference type parameter to `ASON` will result in a runtime error.
-
-- When using the regular `ASON.serialize` method, values automatically will be boxed and unboxed for you.  
-  
-```ts  
-assert(ASON.deserialize<i32>(ASON.serialize(42)) == 42);  
-```  
-
-- However, when using the `Serializer` and `Deserializer` class directly, values must be boxed like this:  
-  
-```ts  
-class Box<T> { constructor(public value: T) {} }  
-  
-let ser = new Serializer<f32>(); // Compile time error!  
-  
-// instead do this  
-let ser = new ASON.Serializer<Box<f32>>();  
-let des = new ASON.Deserializer<Box<f32>>();  
-assert(des.deserialize<Box<f32>>(ser.serialize(new Box<f32>(42))).value == <f32>42);  
-```  
+- If the modules using this library are different, then runtime type information might not match. This will result in runtime errors, `instanceof` checks failing, and undefined behavior. ASON also performs type information validation for objects at the top level, so providing the wrong reference type parameter to the `ASON.deserialize` function  will result in a runtime error.
+
+- Number types will always be boxed for you.
```

Comment:
This isn't a caveat anymore.

Reply:
It's true. Under the hood, we *must* allocate a reference to take advantage of the underlying infrastructure. Values *will* be boxed inside the `Serialize` and `Deserialize` class.

This *is* a caveat.